src/main/scala/fr/acinq/bitcoin/scalacompat/ScriptTree.scala

```diff
@@ -0,0 +1,56 @@
+package fr.acinq.bitcoin.scalacompat
+
+import scodec.bits.ByteVector
+
+/** Simple binary tree structure containing taproot spending scripts. */
+sealed trait ScriptTree {
```

Comment:
We're not using these new ScriptTree wrappers yet in https://github.com/ACINQ/eclair/blob/update-bitcoin-lib/eclair-core/src/main/scala/fr/acinq/eclair/transactions/Scripts.scala  in https://github.com/ACINQ/eclair/pull/3179 it would highlight how they make things simpler